### PR TITLE
build: pass TAGS to test target

### PR DESCRIPTION
**Pull request by yaojingguo**
pass TAGS to `test` target in `Makefile`

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9421)

### CODE REVIEW

**Comment by tamird** [comment]
LGTM but this should go to master and you should at least fix testrace as
well while you're here, and it would be good to get stress and stressrace
too.

**Comment by yaojingguo** [comment]
Pass TAGS to all `go test`-related targets and pass it to `go tool vet` in `build/check-style.sh`.

**Comment by tamird** [comment]
LGTM cc @bdarnell

**Comment by bdarnell** [comment]
LGTM. If we support passing tags into `build` and `install` we should definitely support it for the rest of the commands too. I think we should also be passing `-installsuffix` whenever `-tags` are used, but since `make build` and `make install` are already using `-tags` without `-installsuffix` that's no reason to block this change.

File: Makefile [hunk 1]
```diff
@@ -169,8 +169,8 @@ dupl:
 check:
 	# compile everything; go vet sometimes reports incorrect errors if
 	# the build artifacts are stale.
-	$(GO) test -i ./...
-	@build/check-style.sh
+	$(GO) test -i -tags '$(TAGS)' ./...
+	@build/check-style.sh ''\''$(TAGS)'\'''
```
Comment (bdarnell) [inline Makefile]: This quoting doesn't look right. Why can't it just be '$(TAGS)` like the rest?

File: build/check-style.sh [hunk 2]
```diff
@@ -3,6 +3,7 @@
 set -euo pipefail
 
 export PKG=${PKG:-./...}
+export TAGS=${1:-"''"}
```
Comment (bdarnell) [inline build/check-style.sh]: And why isn't this `${1:-}`? You're quoting the expansion below so extra quotes shouldn't be needed.

File: Makefile (hunk 1 above)
Comment (yaojingguo) [inline Makefile]: Yes, your are right. Fixed in the commit.

File: build/check-style.sh (hunk 2 above)
Comment (yaojingguo) [inline build/check-style.sh]: Fixed in the new commit.

**Comment by yaojingguo** [comment]
It is worthwhile to mention that `go build`, `go test` and `go tool vet` handle tag list in different ways. I have opened an issue https://github.com/golang/go/issues/17148 for it. Here are our make targets corresponding to these go commands:

- `make build TAGS=...` invokes `go build`. 
- `make test TAGS=...` invokes `go test`. 
-  `make check TAGS=...` invokes `go tool vet`. 

So we need to use space-separated tag list for `make build TAGS=...` and `make test TAGS=...`. And for `make check TAGS=...`, we need to use comma-separated tags list. How should we deal with this inconsistency? Document it somewhere? Or make `TAGS` use a consistent format?

**Comment by bdarnell** [comment]
Why do we need to pass tags to `vet`? I think it's OK if `vet` only looks at files in the default build configuration (especially since none of the other linters support tags). We should just use the `go build` format for now and we can add tags to `vet` when it is updated to be consistent with the other tools.

**Comment by yaojingguo** [comment]
@bdarnell OK. So what we are going to do? Is this PR OK for merge? Or should I remove the code which passes TAGS from `Makefile` to `build/check-style.sh`?

Now the CircleCI build fails with such a message. I think that it is not caused by my PR:

```
Your build ran 1151 tests with 1 failure
 TestParallelCreateConflictingTables - github.com/cockroachdb/cockroach/sql
  moreI160918 15:50:44.995102 3449 storage/engine/rocksdb.go:356  opening in memory rocksdb instance
```

**Comment by bdarnell** [comment]
Yes, remove the part related to `check-style` and this is ready to merge. The test failure isn't your fault.

**Comment by yaojingguo** [comment]
- Remove changes made to `build/check-style.sh`.
- Remove the code passing TAGS to `@build/check-style.sh` in `check` target. But leave TAGS for `$(GO) test ...` in `check` target.

**Comment by bdarnell** [comment]
Thanks!

**Comment by yaojingguo** [comment]
@bdarnell Is it possible for you or some member of the stability team to cherry-pick this PR into the `develop` branch? My another PR https://github.com/cockroachdb/cockroach/pull/8867 depends on this PR.

**Pull request merged**